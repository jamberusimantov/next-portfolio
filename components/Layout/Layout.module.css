/* header */

.header {
    background-image: url('/assets/34040.jpg');
    padding: .5rem;
    position: fixed;
    top: 0;
    left: 0;
    width: 100%;
    color: #eaeaea;
    z-index: 10;
}

.header_top {
    display: flex;
    align-items: center;
}

.socialBtns {
    display: flex;
    justify-content: center;
}

.menuBtn {
    display: flex;
    justify-content: center;
    align-items: center;
}

.hamburger {
    flex: 0;
    height: 50px;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
}

.burger {
    width: 2rem;
    height: .25rem;
    margin: .15rem;
    background-color: #eaeaea;
}

.current {
    background-image: url('/assets/34040.jpg');
    color: silver;
    cursor: pointer;
    padding: .5rem;
}

.nav_desktop {
    display: flex;
    justify-content: space-around;
    max-width: 400px;
    margin: auto;
}

.nav_mobile {
    display: flex;
    flex-direction: column-reverse;
    width: 100%;
    text-align: center;
    padding: .5rem;
    color: #333;
    background-color: silver;
    font-weight: 700;
}

.logoContainer {
    flex: 1;
    display: flex;
    justify-content: center;
    align-items: center;
}

.logo {
    cursor: pointer;
    flex: 1;
    display: flex;
    justify-content: center;
    align-items: center;
    font-size: 1.3rem;
    line-height: 1.3;
}

.homeBtn {
    display: flex;
    justify-content: center;
    align-items: center;
}

.hamburger {
    cursor: pointer;
}

.navItem {
    cursor: pointer;
    padding: .5rem;
}

.headerBtn {
    cursor: pointer;
    color: #eaeaea;
    font-size: 1.3rem;
    line-height: 1.3;
    display: inline-flex;
    justify-content: center;
    align-items: center;
    padding: 0 .25rem;
}

.nav_desktop {
    display: none;
}

.hamburger_logo_container {
    flex: 1;
    display: flex;
}

.phoneBtn {
    display: none;
}

.header_githubBtn {
    display: none;
}

.header_linkedinBtn {
    display: none;
}

.header_whatsappBtn {
    display: none;
}

.links {
    display: flex;
    flex-direction: row-reverse;
}

@media (min-width: 1025px) {
    .header {
        padding: .5rem 2rem;
    }
    .nav_desktop {
        max-width: 800px;
    }
}

@media (min-width: 768px) {
    .main_nav {
        flex: 1;
    }
    .nav_desktop {
        display: flex;
    }
    .hamburger_logo_container {
        flex: 0;
    }
    .menuBtn {
        display: none;
    }
    .phoneBtn_mobile {
        display: none;
    }
    .phoneBtn {
        display: flex;
        color: #eaeaea;
        align-items: flex-start;
    }
    .links {
        flex-direction: column;
    }
    .header_bottom {
        display: none;
    }
}

@media (min-width: 500px) {
    .hamburger_logo_container {
        flex-direction: row-reverse;
    }
    .logo {
        flex: 0;
    }
    .logoContainer {
        flex: 0;
    }
    .menuBtn {
        flex: 1;
    }
    .header_githubBtn {
        display: flex;
    }
    .header_linkedinBtn {
        display: flex;
    }
    .header_whatsappBtn {
        display: flex;
    }
}


/* footer */

.footer {
    background-image: url('/assets/34040.jpg');
    padding: .5rem 2.5rem;
    width: 100%;
    min-height: 50px;
    color: #eaeaea;
    display: flex;
    flex-direction: column;
    align-items: center;
    z-index: 10;
}

.footerTop {
    display: flex;
    align-items: center;
}

.footerBottom {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
}

.copyright {
    flex: 1;
    display: flex;
    justify-content: center;
    align-items: center;
    margin: 0;
    text-align: center;
}

.goTOtopBtn {
    cursor: pointer;
    position: fixed;
    bottom: .5rem;
    left: .5rem;
    border-radius: 5px;
    padding: .25rem;
    border: none;
    display: flex;
    justify-content: center;
    align-items: center;
}

.footerBtn {
    cursor: pointer;
    color: #eaeaea;
    font-size: 1.3rem;
    line-height: 1.3;
    display: inline-flex;
    justify-content: center;
    align-items: center;
    padding: 0 .25rem;
}

.footer_phoneBtn {
    margin: .5rem 0;
}

@media (min-width: 500px) {
    .footerBottom {
        display: none;
    }
}