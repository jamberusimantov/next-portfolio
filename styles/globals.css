html,
body {
    padding: 0;
    margin: 0;
    font-family: -apple-system, BlinkMacSystemFont, Segoe UI, Roboto, Oxygen, Ubuntu, Cantarell, Fira Sans, Droid Sans, Helvetica Neue, sans-serif;
}

a {
    color: inherit;
    text-decoration: none;
}

* {
    box-sizing: border-box;
}

.hamburger {
    width: 40px;
    height: 40px;
    border: 0 solid transparent;
    border-radius: 0;
    cursor: pointer;
    display: inline-block;
    padding: 5px;
    background-color: transparent;
    opacity: 1;
}

.inner {
    margin: -1px 0 0;
    position: relative;
    width: 100%
}

.hamburger .bar {
    background-color: var(--white);
    border-radius: 0;
    content: "";
    display: block;
    height: 2px;
    position: absolute;
    width: 100%
}

.hamburger .bar:first-child {
    top: -10px;
}

.hamburger .bar:nth-child(3) {
    top: 10px;
}

.hamburger.active .bar:first-child {
    transform: rotate(45deg) translate3d(7px, 7px, 0);
    transition-delay: .1s
}

.hamburger.active .bar:nth-child(2) {
    transform: scaleX(0);
    transition-delay: .05s
}

.hamburger.active .bar:nth-child(3) {
    transform: rotate(-45deg) translate3d(7px, -7px, 0);
    transition-delay: .1s
}

.page {
    position: relative;
    display: flex;
    flex-direction: column;
    justify-content: center;
    font-size: large;
}

.pageContainer {
    position: relative;
    min-height: calc(100vh - 4.5rem);
}

.submitBtn {
    width: 100%;
    max-width: 150px;
    display: flex;
    justify-content: center;
    align-items: center;
    padding: .5rem;
}

.submitBtn p {
    flex: 1;
    margin: 0;
    font-size: large;
    padding: 0 !important;
}

.response {
    height: 25px;
    width: 25px;
    padding: 0 2rem;
    background-repeat: no-repeat;
    background-position: center;
    background-size: contain;
}

.success {
    background-image: url('/assets/svg/success.svg');
}

.failure {
    background-image: url('/assets/svg/failure.svg');
}

.submit {
    background-image: url('/assets/svg/submit.svg');
}

@media(min-width:580px) {
    .submitBtn {
        margin: auto;
    }
}

@media(min-width:768px) {
    .pageContainer {
        min-height: calc(100vh - 7rem);
    }
}