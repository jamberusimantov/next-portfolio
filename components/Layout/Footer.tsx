import React, { useEffect, useState } from 'react'
import { CustomBtn, Svg } from '../../dir/elements'
import styles from './Layout.module.css'
import { CustomLink } from '../../dir/elements'


const Footer = () => {
  const [isScroll, setIsScroll] = useState(false)

  useEffect(() => {
    const scrollHandler = () => { setIsScroll(scrollY > 50) }
    addEventListener('scroll', scrollHandler);
    return (() => { removeEventListener('scroll', scrollHandler) })
  })

  return (
    <footer className={styles.footer}>
      <div className={styles.footerTop}>
        {isScroll && <CustomBtn
          className={styles.goTOtopBtn}
          child={{
            name: 'up-arrow',
            size: 15,
          }}
          click={() => { scroll({ top: 0, behavior: 'smooth' }) }}
        />
        }
        <div className={styles.copyright}>
          Made with ❤ in Israel © 2022 Siman tov Jamberu. All rights reserved.
        </div>
      </div>
      <div className={styles.footerBottom}>
        <div>
          <CustomLink
            url="tel:+972528405624"
            className={`${styles.footerBtn} ${styles.footer_phoneBtn}`}
            child={{
              className: 'svg_square',
              text: '052-840-5624',
              size: 25
            }} />
          :טלפון
        </div>
        <div>
          <CustomLink
            url='https://api.whatsapp.com/send?phone=+972528405624&text=שלום%20סימן%20טוב%20אשמח%20לדעת%20יותר'
            className={`${styles.footerBtn} ${styles.footer_whatsappBtn}`}
            child={{
              name: 'whatsapp',
              size: 25
            }} />
          <CustomLink
            url='https://github.com/jamberusimantov'
            className={`${styles.footerBtn} ${styles.footer_githubBtn}`}
            child={{
              name: 'github',
              size: 25
            }} />
          <CustomLink
            url='https://www.linkedin.com/in/siman-tov-jamberu/'
            className={`${styles.footerBtn} ${styles.footer_linkedinBtn}`}
            child={{
              name: 'linkedin',
              size: 25
            }} />
        </div>
      </div>
    </footer>
  )
}
export default Footer