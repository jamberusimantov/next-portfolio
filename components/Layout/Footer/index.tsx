import React, { useContext } from 'react';
import styles from './index.module.css';
import CustomBtn from '../../CustomBtn';
import CustomLink from '../../CustomLink';
import dictionary from '../../../dir/dictionary';
import HTMLContext from '../../../contexts/htmlContext';


const Decoration = () => <span className={"decoration"}>&gt;</span>;

const Footer = () => {
  const { setHtml, html } = useContext(HTMLContext);
  return (
    <footer className={styles.footer}>
      <div className={styles.footerTop}>
        <div className={styles.footerTopContainer}>
          <div className={styles.footerLinks}>
            <div className={styles.serviceLinks}>
              <h3>
                {dictionary.services[html.lang]}
              </h3>
              <ul>
                <li>
                  <Decoration />
                  <CustomLink
                    href={"/contact"}
                    title={dictionary.contact["en-US"]}
                    children={dictionary.contact[html.lang]}
                    className="white"
                  />
                </li>
                <li>
                  <Decoration />
                  <CustomLink
                    href={"/help"}
                    title={dictionary.help["en-US"]}
                    children={dictionary.help[html.lang]}
                    className="white"
                  />
                </li>
                <li>
                  <Decoration />
                  <CustomLink
                    href={"/archive"}
                    title={dictionary.archive["en-US"]}
                    children={dictionary.archive[html.lang]}
                    className="white"
                  />
                </li>
              </ul>
            </div>
          </div>
        </div>
      </div>
      <div className={styles.footerBottom}>
        <div className={styles.footerBottomContainer}>
          <div className={styles.footerLang}>
            <div className={styles.lang}>
              <CustomBtn
                text={'English'}
                className={html.lang == "en-US" ? styles.currentLang : ''}
                handleClick={() => setHtml({ lang: 'en-US', dir: 'ltr' })}
              />
            </div>
            <div className={styles.lang}>
              <CustomBtn
                text={'עברית'}
                className={html.lang == "he-IL" ? styles.currentLang : ''}
                handleClick={() => setHtml({ lang: 'he-IL', dir: 'rtl' })}
              />
            </div>
          </div>
        </div>
      </div>
    </footer>
  );
};

export default Footer;