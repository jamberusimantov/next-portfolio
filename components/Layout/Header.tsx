import React, { useState } from 'react'
import { CustomLink, Svg } from '../../dir/elements'
import Link from 'next/link'
import styles from './Layout.module.css'

const NavLinks = () => <>
    <Link
        href='/contact'
        children={
            <span className={styles.navItem}>צור קשר</span>
        }
    />
    <Link
        href='/showcases'
        children={
            <span className={styles.navItem}>עבודות</span>
        }
    />
    <Link
        href='/profile'
        children={
            <span className={styles.navItem}>פרופיל</span>
        }
    />
    <Link
        href='/'
        children={
            <span className={styles.navItem}>דף הבית</span>
        }
    />
</>

const Header = () => {
    const [isMenuOpen, setIsMenuOpen] = useState(false);

    return (
        <header className={styles.header}>
            <div className={styles.header_top}>

                <div className={styles.hamburger_logo_container}>
                    <div className={styles.menuBtn}>
                        <div
                            className={styles.hamburger}
                            onClick={() => { setIsMenuOpen(!isMenuOpen) }}
                        >
                            <div className={`${styles.burger} ${styles.burger1}`}></div>
                            <div className={`${styles.burger} ${styles.burger2}`}></div>
                            <div className={`${styles.burger} ${styles.burger3}`}></div>
                        </div>
                    </div>
                    <div className={styles.logoContainer}>
                        WEB
                        <CustomLink
                            url="/"
                            className={styles.logo}
                            child={{
                                name: 'web',
                                className: 'styles.svg_round',
                                size: 50,
                                samePage: true,
                            }} />
                        DEV
                    </div>
                </div>

                <div className={styles.main_nav}>
                    <nav className={styles.nav_desktop}>
                        <NavLinks />
                    </nav>
                </div>

                <div className={styles.links}>
                    <CustomLink
                        url="tel:+972528405624"
                        className={`${styles.headerBtn} ${styles.phoneBtn}`}
                        child={{
                            name: 'phone',
                            className: 'svg_square',
                            text: '052-840-5624',
                            size: 25
                        }} />
                    <CustomLink
                        url="tel:+972528405624"
                        className={`${styles.headerBtn} ${styles.phoneBtn_mobile}`}
                        child={{
                            name: 'phone',
                            className: 'svg_square',
                            size: 25
                        }} />
                    <div className={styles.socialBtns}>
                        <CustomLink
                            url='https://api.whatsapp.com/send?phone=+972528405624&text=שלום%20סימן%20טוב%20אשמח%20לדעת%20יותר'
                            className={`${styles.headerBtn} ${styles.header_whatsappBtn}`}
                            child={{
                                name: 'whatsapp',
                                size: 25
                            }} />
                        <CustomLink
                            url='https://github.com/jamberusimantov'
                            className={`${styles.headerBtn} ${styles.header_githubBtn}`}
                            child={{
                                name: 'github',
                                size: 25
                            }} />
                        <CustomLink
                            url='https://www.linkedin.com/in/siman-tov-jamberu/'
                            className={`${styles.headerBtn} ${styles.header_linkedinBtn}`}
                            child={{
                                name: 'linkedin',
                                size: 25
                            }} />
                    </div>
                </div>
            </div>

            <div className={`${styles.header_bottom} ${styles['Menu_' + isMenuOpen]}`}>
                <nav className={styles.nav_mobile}>
                    <NavLinks />
                </nav>
            </div>
        </header >
    )
}
export default Header