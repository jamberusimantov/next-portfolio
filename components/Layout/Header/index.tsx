import React from 'react';
import styles from './index.module.css';
import Image from 'next/image';


const Header = () => {
    return (
        <header className={styles.header}>
            <div className={styles.headerTop}>
                <div className={styles.headerTopContainer}>
                    <div className={styles.logo}>
                        <div className={styles.logoContainer}>
                            <Image src='/assets/svg/web-development.svg' layout='fill' />
                        </div>
                        <span className={styles.name}>Siman Tov Jamberu</span>
                    </div>
                </div>
            </div>
        </header >
    );
};

export default Header;