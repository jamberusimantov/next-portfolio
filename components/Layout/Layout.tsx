import Head from 'next/head'
import React, { ReactElement } from 'react'
import Footer from './Footer'
import Header from './Header'

const Layout = (props: { children: ReactElement }) => {

    return (
        <div className='container'>
            <Head>
                <title>Portfolio</title>
                <meta name="application-name" content='Portfolio' />
                <meta name="description" content='just another Portfolio' />
                <meta name="author" content="siman tov jamberu" />
                <meta name="viewport" content="width=device-width, initial-scale=1.0" />
                <link rel="icon" href="/favicon.ico" />
            </Head>
            <Header />
            <main>{props.children}</main>
            <Footer />
        </div>
    )
}

export default Layout