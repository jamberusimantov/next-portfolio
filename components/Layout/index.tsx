import React, { ReactElement, useEffect, useContext } from 'react'
import styles from './Layout.module.css'
import Head from 'next/head'
import Header from './Header'
import Footer from './Footer'
// import { getAppStateHtml } from "../../store/hooks"
import { HTMLContext } from "../../pages/_app"


interface iLayoutProps {
    children: ReactElement
}

const Layout = (props: iLayoutProps) => {
    // const html = getAppStateHtml();
    const { html } = useContext(HTMLContext);
    useEffect(() => {
        const htmlEl = document.querySelector('html')
        if (!htmlEl) return
        htmlEl.dir = html.dir
        htmlEl.lang = html.lang
    }, [html])

    return (
        <div className={styles.layout}>
            <Head>
                <title>Siman Tov Jamberu Portfolio</title>
                <meta name="application-name" content='Portfolio' />
                <meta name="description" content='just another Portfolio' />
                <meta name="author" content="siman tov jamberu" />
                <meta name="viewport" content="width=device-width, initial-scale=1.0" />
                <link rel="icon" href="/favicon.ico" />
            </Head>
            <Header />
            <main children={props.children} />
            <Footer />
        </div>
    )
}

export default Layout