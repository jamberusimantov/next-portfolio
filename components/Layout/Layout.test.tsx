/**
 * @jest-environment jsdom
 */
import React from "react";
import { describe, expect, test } from '@jest/globals';
import { render,act } from '@testing-library/react';
import Layout from '.';


describe('Layout', () => {
    test('use jsdom in this test file', () => {
        expect(document.createElement('div')).not.toBeNull();
    });
    test('renders correctly', async () => {
        let el: HTMLElement = document.createElement("div");
        let getText: Function = () => { };
        await act(async () => {
            let {container,getByText} = await render(<Layout children={ <div>I'm div wrapped with layout</div>}/>);
            el = container;
            getText = getByText;
        });
        expect(el).toMatchSnapshot();
        expect(getText("I'm div wrapped with layout"));
        expect(getText("English"));
        expect(getText("עברית"));
    });
});