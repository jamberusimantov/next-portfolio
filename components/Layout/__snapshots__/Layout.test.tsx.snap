// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders correctly 1`] = `
<div>
  <div
    class="layout"
  >
    <header
      class="header"
    >
      <div
        class="headerTop"
      >
        <div
          class="headerTopContainer"
        >
          <div
            class="logo"
          >
            <div
              class="logoContainer"
            >
              <span
                style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
              >
                <img
                  data-nimg="fill"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                />
                <noscript />
              </span>
            </div>
            <span
              class="name"
            >
              Siman Tov Jamberu
            </span>
          </div>
        </div>
      </div>
    </header>
    <main>
      <div>
        I'm div wrapped with layout
      </div>
    </main>
    <footer
      class="footer"
    >
      <div
        class="footerTop"
      >
        <div
          class="footerTopContainer"
        >
          <div
            class="footerLinks"
          >
            <div
              class="serviceLinks"
            >
              <h3>
                שירותים
              </h3>
              <ul>
                <li>
                  <span
                    class="decoration"
                  >
                    &gt;
                  </span>
                  <a
                    class="white"
                    href="/contact"
                    title="Contact"
                  >
                    צור קשר
                  </a>
                </li>
                <li>
                  <span
                    class="decoration"
                  >
                    &gt;
                  </span>
                  <a
                    class="white"
                    href="/help"
                    title="Help"
                  >
                    עזרה
                  </a>
                </li>
                <li>
                  <span
                    class="decoration"
                  >
                    &gt;
                  </span>
                  <a
                    class="white"
                    href="/archive"
                    title="Archive"
                  >
                    ארכיון
                  </a>
                </li>
              </ul>
            </div>
          </div>
        </div>
      </div>
      <div
        class="footerBottom"
      >
        <div
          class="footerBottomContainer"
        >
          <div
            class="footerLang"
          >
            <div
              class="lang"
            >
              <button
                class=" button"
                type="button"
              >
                <p>
                  English
                </p>
              </button>
            </div>
            <div
              class="lang"
            >
              <button
                class="currentLang button"
                type="button"
              >
                <p>
                  עברית
                </p>
              </button>
            </div>
          </div>
        </div>
      </div>
    </footer>
  </div>
</div>
`;
